src/lib/baseCommand.ts

```diff
@@ -11,11 +10,16 @@ import isCI from './isCI';
 import { debug, info, warn } from './logger';
 import loginFlow from './loginFlow';
 
-export type CommandOptions<T> = T & {
-  github?: boolean;
-  key?: string;
+export type CommandOptions<T> = ZeroAuthCommandOptions<T> | AuthenticatedCommandOptions<T>;
+
+export type AuthenticatedCommandOptions<T> = Omit<ZeroAuthCommandOptions<T>, 'key'> & {
+  key: string;
   version?: string;
-} & CommandLineOptions;
+};
+
+export type ZeroAuthCommandOptions<T> = T & {
```

Comment:
Since we pass in `<{}>` to this generic in places where we don't have any options do you want to default `T` to that?

```suggestion
export type ZeroAuthCommandOptions<T = {}> = T & {
```